src/fmu/dataio/providers/objectdata/_base.py

```diff
@@ -270,72 +334,24 @@ def get_bbox(self) -> BoundingBox2D | BoundingBox3D | None:
     def get_objectdata(self) -> DerivedObjectDescriptor:
         raise NotImplementedError
 
-    def derive_metadata(self) -> None:
-        """Main function here, will populate the metadata block for 'data'."""
-        logger.info("Derive all metadata for data object...")
-
-        namedstratigraphy = self._derive_name_stratigraphy()
-        objres = self.get_objectdata()
-        content_model = get_validated_content(self.dataio.content)
-
-        if self.dataio.forcefolder:
-            if self.dataio.forcefolder.startswith("/"):
-                raise ValueError("Can't use absolute path as 'forcefolder'")
-            msg = (
-                f"The standard folder name is overrided from {objres.efolder} to "
-                f"{self.dataio.forcefolder}"
-            )
-            objres.efolder = self.dataio.forcefolder
-            logger.info(msg)
-            warn(msg, UserWarning)
-
-        meta = self.metadata  # shortform
-
-        meta["name"] = namedstratigraphy.name
-        meta["stratigraphic"] = namedstratigraphy.stratigraphic
-        meta["offset"] = namedstratigraphy.offset
-        meta["alias"] = namedstratigraphy.alias
-        meta["top"] = namedstratigraphy.top
-        meta["base"] = namedstratigraphy.base
+    def get_metadata(self) -> AnyContent | UnsetAnyContent:
+        return (
+            UnsetAnyContent.model_validate(self.metadata)
+            if self.metadata["content"] == "unset"
+            else AnyContent.model_validate(self.metadata)
+        )
```

Comment:
This is the only new functionality in this class, aside from deriving from `Provider` now.